index.js

```diff
@@ -17,41 +17,56 @@ const DEFAULT_ALLOW_HEADERS = [
 
 const DEFAULT_MAX_AGE_SECONDS = 60 * 60 * 24 // 24 hours
 
-const cors = options => handler => (req, res) => {
+const checkOrigin = (origin, allowed) => {
+  if (Array.isArray(allowed)) return !!allowed.find(a => a === origin)
+  if (typeof allowed === 'function') return allowed(origin)
+  if (allowed instanceof RegExp) return allowed.test(origin)
+  return !!allowed
```

Comment:
For the most part cribbed directly from `express-cors` - figured it would be handy for folks coming over from express to have their cors definitions "just work" with this lib.